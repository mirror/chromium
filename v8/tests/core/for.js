print("simple for loop");
for (var i = 0; i < 10; i++)
  print(i);

print("simple for loop w/ break");
for (var i = 0; i < 10; i++) {
  if (i == 7)
    break;
  print(i);
}

print("simple for loop w/ break & continue");
for (var i = 0; i < 10; i++) {
  if (i == 3)
    continue;
  if (i == 7)
    break;
  print(i);
}

print("simple for loop w/ break & continue, using a nested switch");
loop: for (var i = 0; i < 10; i++) {
  switch (i) {
    case 3: continue;
    case 7: break loop;
  }
  print(i);
}

print("endless for loop");
for (var i = 0; ; i--) {
  print(i);
  if (i < -5)
    break;
}
