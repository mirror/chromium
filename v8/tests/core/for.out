simple for loop
0
1
2
3
4
5
6
7
8
9
simple for loop w/ break
0
1
2
3
4
5
6
simple for loop w/ break & continue
0
1
2
4
5
6
simple for loop w/ break & continue, using a nested switch
0
1
2
4
5
6
endless for loop
0
-1
-2
-3
-4
-5
-6
